presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -478,28 +485,75 @@ private void testRoundTripType(Type type, List<?> readValues)
 
     public void testRoundTripTypes(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters)
             throws Exception
+    {
+        testRoundTripTypes(types, readValues, filters, Optional.empty());
+    }
+
+    public void testRoundTripTypes(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters, Optional<List> order)
+            throws Exception
     {
         assertEquals(types.size(), readValues.size());
+        Optional<OrderChecker> optionalOrderChecker = Optional.empty();
```

Comment:
Remove `setDesiredOrder` and pass expected order to the constructor of the OrderChecker:

```
Optional<OrderChecker> optionalOrderChecker = order.map(OrderChecker::new);

        private final List<Integer> expectedOrder;
        private final List<Integer> columnOrdering = new ArrayList<>();
        private boolean started;
        private boolean ended;

        OrderChecker(List<Integer> expectedOrder)
        {
            this.expectedOrder = requireNonNull(expectedOrder, "expectedOrder is null");
        }
```